packages/core/src/baseclient.ts

```diff
@@ -280,21 +286,25 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
   public sendSession(session: Session | SessionAggregates): void {
     if (this._dsn) {
       const env = createSessionEnvelope(session, this._dsn, this._options._metadata, this._options.tunnel);
-      this.sendEnvelope(env);
+      this._sendEnvelope(env);
     }
   }
 
   /**
-   * @inheritdoc
+   * @inheritDoc
    */
-  public sendEnvelope(env: Envelope): void {
-    if (this._transport && this._dsn) {
-      this._transport.send(env).then(null, reason => {
-        IS_DEBUG_BUILD && logger.error('Error while sending event:', reason);
-      });
-    } else {
-      IS_DEBUG_BUILD && logger.error('Transport disabled');
-    }
+  public recordDroppedEvent(reason: EventDropReason, category: DataCategory): void {
```

Comment:
This should only work if the `sendClientReports` option is true.

Reply:
> I think we can get away with recording the events, but only actually sending them when sendClientReports is true

I was thinking problems with integer overflow or outcomes for long running applications (like a node server), but maybe not that big of a deal with what you showed before.